src/main/java/gmbs/game/BaseballRuleChecker.java

```diff
@@ -0,0 +1,58 @@
+package gmbs.game;
+
+import gmbs.common.Constant;
+
+import java.util.List;
+import java.util.function.UnaryOperator;
+import java.util.stream.Stream;
+
+public class BaseballRuleChecker {
+
+    private static final int START_INDEX = 0;
+
+    private final int strikeCount;
+    private final int ballCount;
+
+    public BaseballRuleChecker(List<Integer> randomNumberList, List<Integer> inputNumberList) {
+        this.strikeCount = setStrikeCount(randomNumberList, inputNumberList);
+        this.ballCount = setBallCount(randomNumberList, inputNumberList);
+    }
+
+    private int setStrikeCount(List<Integer> randomNumberList, List<Integer> inputNumberList) {
+         return (int) Stream.iterate(START_INDEX, increaseOneOperator()).limit(Constant.MAX_INPUT_SIZE)
+                .filter(index -> isSameValueAtSameIndex(randomNumberList, inputNumberList, index))
+                .count();
+    }
+
+    private int setBallCount(List<Integer> randomNumberList, List<Integer> inputNumberList) {
+        return (int) Stream.iterate(START_INDEX, increaseOneOperator()).limit(Constant.MAX_INPUT_SIZE)
+                .filter(index ->
+                        !isSameValueAtSameIndex(randomNumberList, inputNumberList, index)
+                        && isContainValue(randomNumberList, inputNumberList, index))
+                .count();
+    }
+
+    private UnaryOperator<Integer> increaseOneOperator() {
+        return index -> index + 1;
+    }
+
+    private boolean isSameValueAtSameIndex(List<Integer> randomNumberList, List<Integer> inputNumberList, Integer index) {
+        return randomNumberList.get(index).equals(inputNumberList.get(index));
```

Comment:
객체지향 생활 체조 4번, 한 줄에 점을 하나만 찍는다. 
체이닝 되는 메서드들 사용에서는 두개 이상의 점 부터 개행을 해주는게 좋아요~ 
```suggestion
        return randomNumberList.get(index)
                .equals(inputNumberList.get(index));
```